pkg/placement/placement.go

```diff
@@ -142,29 +161,55 @@ func (c *Controller) requeueAll() error {
 	return nil
 }
 
-// finalizer logic
-func (c *Controller) handlePlacementFinalizer(obj runtime.Object) error {
-	mObj := obj.(metav1.Object)
-	cObj, err := convertToClientObject(obj)
+func (c *Controller) queuePlacementDecision(name string) error {
+	placementDecisions, err := c.listPlacementDecisions()
```

Comment:
At this point it is already known what `util.KeyForGroupVersionKindNamespaceName` would return, we do not need to get an actual object in hand.